boards/raspberrypi/rpi_pico2/doc/index.rst

```diff
@@ -0,0 +1,92 @@
+.. _rpi_pico2:
+
+Raspberry Pi Pico 2
+###################
+
+Overview
+********
+
+The Raspberry Pi Pico 2 is the second-generation product in the Raspberry Pi
+Pico family. From the `Raspberry Pi website <https://www.raspberrypi.com/documentation/microcontrollers/pico-series.html>`_ is referred to as Pico 2.
+
+.. figure:: img/pico-2.jpg
+   :align: center
+   :alt: Raspberry Pi Pico 2
+
+   Raspberry Pi Pico 2 (Credit: Raspberry Pi Ltd)
+
+Hardware
+********
+
+- Dual Cortex-M33 or Hazard3 processors at up to 150MHz
+- 520KB of SRAM, and 4MB of on-board flash memory
+- USB 1.1 with device and host support
+- Low-power sleep and dormant modes
+- Drag-and-drop programming using mass storage over USB
+- 26 multi-function GPIO pins including 3 that can be used for ADC
+- 2 SPI, 2 I2C, 2 UART, 3 12-bit 500ksps Analogue to Digital - Converter (ADC), 24 controllable PWM channels
+- 2 Timer with 4 alarms, 1 AON Timer
+- Temperature sensor
+- 3 Programmable IO (PIO) blocks, 12 state machines total for custom peripheral support
+   - Flexible, user-programmable high-speed IO
+   - Can emulate interfaces such as SD Card and VGA
+
+Supported Features
+==================
+
+The rpi_pico board configuration supports the following
+hardware features:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Peripheral
+     - Kconfig option
+     - Devicetree compatible
+   * - NVIC
+     - N/A
+     - :dtcompatible:`arm,v8m-nvic`
+   * - ADC
+     - :kconfig:option:`CONFIG_ADC`
+     - :dtcompatible:`raspberrypi,pico-adc`
+   * - Clock controller
+     - :kconfig:option:`CONFIG_CLOCK_CONTROL`
+     - :dtcompatible:`raspberrypi,pico-clock-controller`
+   * - Counter
+     - :kconfig:option:`CONFIG_COUNTER`
+     - :dtcompatible:`raspberrypi,pico-timer`
+   * - GPIO
+     - :kconfig:option:`CONFIG_GPIO`
+     - :dtcompatible:`raspberrypi,pico-gpio`
+   * - HWINFO
+     - :kconfig:option:`CONFIG_HWINFO`
+     - N/A
+   * - I2C
+     - :kconfig:option:`CONFIG_I2C`
+     - :dtcompatible:`snps,designware-i2c`
+   * - PWM
+     - :kconfig:option:`CONFIG_PWM`
+     - :dtcompatible:`raspberrypi,pico-pwm`
+   * - SPI
+     - :kconfig:option:`CONFIG_SPI`
+     - :dtcompatible:`raspberrypi,pico-spi`
+   * - UART
+     - :kconfig:option:`CONFIG_SERIAL`
+     - :dtcompatible:`raspberrypi,pico-uart`
+
+Connections and IOs
+===================
+
+The default pin mapping is unchanged from the Pico 1.
```

Comment:
it should be listed here, if someone gets this board and has never used the other one, why do they need to flip between documentation for a board they have and documentation for a board they don't have and never did have?

Reply:
> it should be listed here

I disagree.

> if someone gets this board and has never used the other one, why do they need to flip between documentation for a board they have and documentation for a board they don't have and never did have?

It's a reasonable expectation that the reader is comfortable navigating the documentation within the Zephyr codebase, following references to the primary sources listed under references, etc. The benefit of such brevity also allows the reader to determine that they can use the Pico 2 configuration as-is and it will be drop-in compatible with a project using a Pico 1

Reply:
@kartben

Reply:
IMO, we shouldn't assume that Pico2 users are also using Pico1, and the two should be documented as separate entities. I agree with @nordicjm opinion.

Reply:
Is someone going to provide a commit to cherry-pick? Politely, that's creating more work for me.

Reply:
This is your PR? You are responsible for addressing review comments

Reply:
I'm with Jamie, this comment should go, in general (and not about this PR specifically) there's way too much noise in the board docs.

Reply:
The documentation preview for this PR is at [docs/boards/raspberrypi/rpi_pico2/doc/index.html#connections-and-ios](https://builds.zephyrproject.io/zephyr/pr/77368/docs/boards/raspberrypi/rpi_pico2/doc/index.html#connections-and-ios) . It now includes a link that takes the reader directly to the required section in the Pico 1's documentation at [docs/boards/raspberrypi/rpi_pico/doc/index.html#rpi-pico-pin-mapping](https://builds.zephyrproject.io/zephyr/pr/77368/docs/boards/raspberrypi/rpi_pico/doc/index.html#rpi-pico-pin-mapping).

I feel that this is sufficient given a reasonable assumption about the abilities of the reader. Can we agree that this is not a blocker for this PR to be accepted? @kartben

Reply:
I do not agree, the information should be present in the file. The documentation should be self contained.

Reply:
I personally don't mind the cross referencing, especially when it's meant to avoid duplication. That being said, it should be pretty straightforward to *include* the very same bits from the Pico into the Pico 2 document

```
.. literalinclude:: relativepath/to/pico1/index.rst
   :start-after: pico1_pin_mapping_begin
   :end-before: pico1_pin_mapping_end
```

And in the pico1 doc you'd basically enclose the pin mapping section with the two markers, as ReST comments

```
[ ... ]

.. pico1_pin_mapping_begin

[ ... ]

.. pico1_pin_mapping_end

```

Reply:
I tend to be with Jamie, not worth a fight on this though so whatever.

In general I don't see much point in having the project maintaining this giant blurb of freetext, half of it copypasted from brochures that should really live and die in the vendor website.

@kartben @gmarull what if we were to move in the direction of having those pages generated? I'm thinking that there could be a list of image files in `board.yml`, a list of devices instantiated in the board dts file (can we parse them an generate the doc from that? linking to the binding docs) and a list of flasher instruction depenidng on what's defined in `board.cmake`, and then MAYBE a freetext "note" section from the board.yml file.

Reply:
> @kartben @gmarull what if we were to move in the direction of having those pages generated? I'm thinking that there could be a list of image files in `board.yml`, a list of devices instantiated in the board dts file (can we parse them an generate the doc from that? linking to the binding docs) and a list of flasher instruction depenidng on what's defined in `board.cmake`, and then MAYBE a freetext "note" section from the board.yml file.

@fabiobaltieri 
Ya, already on it, sir :)
https://builds.zephyrproject.io/zephyr/pr/79915/docs/boards/seeed/wio_terminal/doc/index.html (see "identity card" on the right hand side)
https://builds.zephyrproject.io/zephyr/pr/79915/docs/boards/seeed/wio_terminal/doc/index.html#supported-features or https://builds.zephyrproject.io/zephyr/pr/79915/docs/boards/seeed/xiao_esp32s3/doc/index.html#supported-features (still need to surface the capabilities per board target, but you get the idea :))

Reply:
@nordicjm @kartben @fabiobaltieri 

Automatic table generation will be implemented soon, so I think it's fine to leave it as it is. What do you all think?